app/controllers/submission_assets/renames_controller.rb

```diff
@@ -0,0 +1,48 @@
+class SubmissionAssets::RenamesController < ApplicationController
+  include EventSourcing
+  before_action :set_context
+  def new
+    @rename = SubmissionAsset::Rename.new(
+      submission_asset_id: @submission_asset.id,
+      renamed_at: Time.now.strftime("%Y-%m-%d %H:%M"),
+      filename_old:  @submission_asset.filename,
+      renamed_by_id: current_account.id
+      )
+    @rename.submission_asset = @submission_asset
```

Comment:
Setting `submission_asset_id` and `submission_asset` seems a bit redundant. Please only set the `submission_asset` if possible, then access the id by calling `submission_asset.id` inside the `Rename` class.